scripts/extensions/sams/src/api/sets.ts

```diff
@@ -0,0 +1,104 @@
+import {ISuperdesk, IRestApiResponse} from 'superdesk-api';
+import {ISet, ISetItem} from '../interfaces';
+import {EVENTS} from '../constants';
+
+const RESOURCE = 'sams/sets';
+
+export function getAllSets(superdesk: ISuperdesk): Promise<Array<ISetItem>> {
+    return superdesk.dataApi.query<ISetItem>(
+        RESOURCE,
+        1,
+        {field: 'name', direction: 'ascending'},
+        {},
+    )
+        .then((response: IRestApiResponse<ISetItem>) => {
+            return response?._items ?? [];
+        });
+}
+
+export function createSet(superdesk: ISuperdesk, item: ISet): Promise<ISetItem> {
+    const {gettext} = superdesk.localization;
+    const {notify} = superdesk.ui;
+
+    return superdesk.dataApi.create<ISetItem>(RESOURCE, item as ISetItem)
+        .then((set: ISetItem) => {
+            notify.success(gettext('Set created successfully'));
+
+            window.dispatchEvent(
+                new CustomEvent(
+                    EVENTS.SET_CREATED,
+                    {detail: set},
+                ),
+            );
+
+            return set;
+        })
+        .catch((error: any) => {
+            notify.error(gettext('Failed to create the Set'));
+
+            throw new Error(error);
```

Comment:
The error response from the API are not currently using a well defined schema. Once this is done we can improve the user experience when it comes to errors or failures of the system (these errors could be more than validation failures)

Reply:
The problem is that there are multiple different ways the libraries, Flask/Eve/Cerberus, returns errors. So we decided to tackle this problem in a separate ticket, rather than write some code now and change that code again in the future

Reply:
Which ticket is that?